arroyo/processing/strategies/decoder/json.py

```diff
@@ -1,26 +1,63 @@
+import logging
+import time
 from typing import Optional
 
 import fastjsonschema  # pip install sentry-arroyo[json]
 import rapidjson
 
 from arroyo.processing.strategies.decoder.base import Codec, ValidationError
+from arroyo.utils.metrics import get_metrics
+
+logger = logging.getLogger(__name__)
 
 
 class JsonCodec(Codec[object]):
-    def __init__(self, json_schema: Optional[object] = None) -> None:
+    def __init__(
+        self,
+        json_schema: Optional[object] = None,
+        json_schema_name: Optional[str] = None,
+        json_schema_raise_failures: bool = True,
+    ) -> None:
+        self.__metrics = get_metrics()
+        self.__json_schema_raise_failures = json_schema_raise_failures
+        self.__json_schema_name = json_schema_name
+
         if json_schema is not None:
             self.__validate = fastjsonschema.compile(json_schema)
         else:
             self.__validate = lambda _: _
 
     def decode(self, raw_data: bytes, validate: bool) -> object:
+        start = time.time()
         decoded = rapidjson.loads(raw_data)
+
+        after_decoded = time.time()
+
+        self.__metrics.timing(
+            "arroyo.strategies.decoder.json.loads",
+            after_decoded - start,
+        )
```

Comment:
This is emitting a metric on every message, which can affect performance. Can we find a way to send this metric say once per second or so?

Reply:
can't think of a better place to emit this metric, or to do it in batches, so I just put this behind sampling.

Reply:
oh, sorry, i just saw the other message. will use that